src/integrationTest/java/uk/gov/hmcts/reform/professionalapi/AddPaymentAccountsIntegrationTest.java

```diff
@@ -0,0 +1,606 @@
+        paymentAccountsEmptyOrNullToAdd.add("");
+        paymentAccountsEmptyOrNullToAdd.add(" ");
+
+        PbaAddRequest pbaAddRequest = new PbaAddRequest();
+        pbaAddRequest.setPaymentAccounts(paymentAccountsToAdd);
+        pbaAddRequest.getPaymentAccounts().addAll(paymentAccountsEmptyOrNullToAdd);
+
+        String userId = createActiveUserAndOrganisation(true);
+
+        Map<String, Object> pbaResponse = professionalReferenceDataClient
+                .addPaymentsAccountsByOrgId(pbaAddRequest, puiFinanceManager, userId);
+
+        LinkedHashMap reason = (LinkedHashMap)pbaResponse.get("reason");
+        ArrayList duplicatePaymentAccountsList = (ArrayList) reason.get("duplicatePaymentAccounts");
+        ArrayList invalidPaymentAccountsList = (ArrayList) reason.get("invalidPaymentAccounts");
+        Object message = pbaResponse.get("message");
+
+        assertThat(message).isEqualTo(ERROR_MSG_PARTIAL_SUCCESS);
+        assertThat(duplicatePaymentAccountsList).contains("PBA0000001", "PBA0000003");
+        assertThat(invalidPaymentAccountsList).contains("invalid-test01","invalid-test02");
+        assertThat(pbaResponse).containsEntry("http_status", "201 CREATED");
+
+        java.util.Map<String, Object> retrievePaymentAccountsByEmailResponse = professionalReferenceDataClient
+                .findPaymentAccountsByEmailFromHeader("someone@somewhere.com", hmctsAdmin);
+
+        Map<String, Object> organisationEntityResponse =
+                (Map<String, Object>) retrievePaymentAccountsByEmailResponse.get("organisationEntityResponse");
+        List<String> paymentAccount = (List<String>) organisationEntityResponse.get("paymentAccount");
+        assertThat(paymentAccount)
+                .hasSize(8)
+                .contains("PBA0000001","PBA0000001","PBA0000001");
+    }
+
+    @Test
+    @SuppressWarnings("unchecked")
+    public void test_add_PaymentAccounts_Failure_Has_Single_emptySpace_PBA_ShouldThrow_400() {
+
+        String request = "{\n"
+                + "    \"paymentAccounts\": [\n"
+                + "        \"  \"\n"
+                + "    ]\n"
+                + "}";
+
+        PbaAddRequest pbaAddRequest = convertJsonReqStringToObj(request);
+
+        String userId = createActiveUserAndOrganisation(true);
+
+        Map<String, Object> pbaResponse = professionalReferenceDataClient
+                .addPaymentsAccountsByOrgId(pbaAddRequest, puiFinanceManager, userId);
+
+        ErrorResponse errorResponse = convertJsonToErrorResponseObj(pbaResponse);
+        assertThat(pbaResponse).containsEntry("http_status", "400");
+        assertThat(errorResponse.getErrorDescription()).isEqualTo(ADD_PBA_REQUEST_EMPTY);
+        assertThat(errorResponse.getErrorMessage()).isEqualTo(INVALID_REQUEST.getErrorMessage());
+    }
+
+    @Test
+    @SuppressWarnings("unchecked")
+    public void test_add_PaymentAccounts_PartialSuccess_Dupl_And_null_02_ShouldReturn_201() {
```

Comment:
The test method name isn't matching with the actual test being done in the method. Please correct the name.